changelog/@unreleased/pr-5940.v2.yml

```diff
@@ -0,0 +1,6 @@
+type: fix
+fix:
+  description: Fixed a race condition where if a transaction was just committed by
+    another thread, we may return null values from TransactionService.get.
```

Comment:
Let's probably say what this means to a user: users may observe that a transaction that had _definitively_ committed or aborted was still running. Retrying the transaction should work.